xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/server/ServerInfoLedger.java

```diff
@@ -93,4 +124,32 @@ static ImmutableServerInfoLedger.Builder builder() {
   @JsonProperty("base_fee_xrp")
   BigDecimal baseFeeXrp();
 
+  /**
+   * Deserializes either scientific notation or decimal notation in JSON to an XrpCurrencyAmount.
+   */
+  class DecimalXrpCurrencyAmountDeserializer extends JsonDeserializer<XrpCurrencyAmount> {
+    @Override
+    public XrpCurrencyAmount deserialize(JsonParser jsonParser, DeserializationContext ctxt) throws IOException {
+      JsonNode node = jsonParser.getCodec().readTree(jsonParser);
+      return XrpCurrencyAmount.ofXrp(new BigDecimal(node.asText()));
+    }
+  }
+
+  /**
+   * Serializes XrpCurrencyAmount in drops to Xrp value as {@link UnsignedInteger}.
+   */
+  class DecimalXrpCurrencyAmountSerializer extends StdScalarSerializer<XrpCurrencyAmount> {
+
+    public DecimalXrpCurrencyAmountSerializer() {
+      super(XrpCurrencyAmount.class, false);
+    }
+
+    @Override
+    public void serialize(XrpCurrencyAmount amount, JsonGenerator gen, SerializerProvider provider) throws IOException {
+      gen.writeNumber(
+        XrpCurrencyAmount.ofDrops(Long.parseLong(amount.toString())).toXrp().setScale(0, RoundingMode.CEILING)
```

Comment:
Couple things here -- 

1) Isn't `XrpCurrencyAmount.ofDrops(Long.parseLong(amount.toString())).toXrp()` just a long way of saying `amount.toXrp()` or am I missing something?
2) Why are we rounding here? The JSON has a decimal, right?